speech/src/main/java/com/example/speech/InfiniteStreamRecognize.java

```diff
@@ -293,10 +218,110 @@ public void onError(Throwable t) {}
 
           clientStream.send(request);
         }
+        clientStream.closeSend();
       } catch (Exception e) {
         System.out.println(e);
       }
     }
   }
+
+  public static ResponseObserver<StreamingRecognizeResponse> getResponseObserver() {
+    return new ResponseObserver<StreamingRecognizeResponse>() {
+
+      final ArrayList<StreamingRecognizeResponse> responses = new ArrayList<>();
+
+      public void onStart(StreamController controller) {
+        referenceToStreamController = controller;
+      }
+
+      public void onResponse(StreamingRecognizeResponse response) {
+        responses.add(response);
+        StreamingRecognitionResult result = response.getResultsList().get(0);
+        Duration resultEndTime = result.getResultEndTime();
+        resultEndTimeInMS =
+                (int) ((resultEndTime.getSeconds() * 1000) + (resultEndTime.getNanos() / 1000000));
+        double correctedTime =
+                resultEndTimeInMS - bridgingOffset + (STREAMING_LIMIT * restartCounter);
+
+        SpeechRecognitionAlternative alternative = result.getAlternativesList().get(0);
+        if (result.getIsFinal()) {
+          System.out.print(GREEN);
+          System.out.print("\033[2K\r");
+          System.out.printf(
+                  "%s: %s [confidence: %.2f]\n",
+                  convertMillisToDate(correctedTime),
+                  alternative.getTranscript(),
+                  alternative.getConfidence());
+          isFinalEndTime = resultEndTimeInMS;
+          lastTranscriptWasFinal = true;
+        } else {
+          System.out.print(RED);
+          System.out.print("\033[2K\r");
+          System.out.printf(
+                  "%s: %s", convertMillisToDate(correctedTime), alternative.getTranscript());
+          lastTranscriptWasFinal = false;
```

Comment:
this is a bad practice for code samples. because

* it leads to flaky tests because it requires parsing captured `stdout` output which is not deterministic.
* it does not add to verbosity of code sample because not everyone who looks into the code will run it to check the actual output
* it creates overhead in understanding these lines while they do not contribute to the code sample

If these prints do not have the meaning in the scope of building the response observer, please remove it. If they serve some role, please refactor the code and make it more verbose and easy to understand. For example, consider using a format template and give it self-explaining name.